### PR TITLE
Use mamba

### PR DESCRIPTION
**Pull request by bryates**
After a _lot_ of debugging, I got the CI working with mamba. It ran in under 2 minutes. For now I kept the original conda version in. We could discuss removing it, since the point of trying mamba was to speed up the CI time.

### CODE REVIEW

**Comment by bryates** [comment]
@kmohrman @Andrew42 @sscruz this is an older, non-critical, PR that added mamba to the CI. Now that the GitHub conda interface is faster, it's not a huge improvement, but might still be nice since some of us (at least me) are using mamba now. This still keeps conda in the CI, and runs a mamba version in parallel.

**Comment by bryates** [comment]
> Does it do the test with full test with both mamba and conda? I am still using conda. Unless we get to a point when everyone is using exclusively mamba, I would hesitate to drop the test with conda.

Yes, it runs both in parallel.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -22,11 +22,50 @@ jobs:
...
+          pytest tests/test_HistEFT_add.py
+        shell: micromamba-shell {0}
+
+
   Check-topcoffea:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
```
[the remark this answers is not in the document]
Comment (bryates) [reply .github/workflows/main.yml:68]: It's updating to the latest GitHub actions (v2 is out of date).

**Comment by kmohrman** [comment]
Ok, I'm happy for this to be merged.

**Comment by bryates** [comment]
> Ok, I'm happy for this to be merged.

Thanks! @Andrew42 @sscruz any comments on this?

**Comment by kmohrman** [comment]
@bryates should we go ahead and merge this? It does not sound like there are any objections.

**Comment by bryates** [comment]
> @bryates should we go ahead and merge this? It does not sound like there are any objections.

Sure, I think this has been sitting long enough.

**Pull request merged**